inc/Engine/Optimization/DelayJS/Admin/Settings.php

```diff
@@ -1,253 +1,95 @@
+			1 === (int) $options['delay_js']
 		) {
-			$delay_js_scripts['pixel-caffeine/build/frontend.js'] = '';
-		}
-
-		if ( isset( $delay_js_scripts['google.com/recaptcha/api.js'] ) ) {
-			unset( $delay_js_scripts['google.com/recaptcha/api.js'] );
+			$options['delay_js_exclusions'] = [
+				$this->get_excluded_internal_paths(),
+				'/jquery-?[0-9.]*(.min|.slim|.slim.min)?.js',
+			];
 		}
 
-		if ( isset( $delay_js_scripts['widget.trustpilot.com'] ) ) {
-			$delay_js_scripts['widget.trustpilot.com/bootstrap'] = $delay_js_scripts['widget.trustpilot.com'];
-
-			unset( $delay_js_scripts['widget.trustpilot.com'] );
-		}
-
-		$options['delay_js_scripts'] = array_values( array_flip( $delay_js_scripts ) );
-
 		update_option( 'wp_rocket_settings', $options );
 	}
 
 	/**
-	 * Restores the default list when updating from 3.7.6 (which removed anything ending in '.js' -- whoops!)
+	 * Sanitizes delay JS options when saving the settings
 	 *
-	 * @since 3.7.6.1
+	 * @since 3.9
 	 *
-	 * @param string $old_version Old plugin version.
+	 * @param array         $input    Array of values submitted from the form.
+	 * @param AdminSettings $settings Settings class instance.
 	 *
-	 * @return void
+	 * @return array
 	 */
-	public function option_update_3_7_6_1( $old_version ) {
-		if ( 0 !== version_compare( $old_version, '3.7.6' ) ) {
-			return;
-		}
-
-		$options = get_option( 'wp_rocket_settings', [] );
-
-		if ( ! isset( $options['delay_js_scripts'] ) || ! is_array( $options['delay_js_scripts'] ) ) {
-			$options['delay_js_scripts'] = $this->defaults;
-		} else {
-			$delay_js_scripts = array_flip( $options['delay_js_scripts'] );
-
-			if ( isset( $delay_js_scripts['a.omappapi.com/app/js/api.min.js'] ) ) {
-				unset( $delay_js_scripts['a.omappapi.com/app/js/api.min.js'] );
-			}
+	public function sanitize_options( $input, $settings ) : array {
+		$input['delay_js']            = $settings->sanitize_checkbox( $input, 'delay_js' );
+		$input['delay_js_exclusions'] = ! empty( $input['delay_js_exclusions'] ) ? rocket_sanitize_textarea_field( 'delay_js_exclusions', $input['delay_js_exclusions'] ) : [];
 
-			if ( isset( $delay_js_scripts['/sdk.js'] ) ) {
-				unset( $delay_js_scripts['/sdk.js'] );
-			}
-
-			$options['delay_js_scripts'] = array_values( array_unique( array_merge( $this->defaults, array_flip( $delay_js_scripts ) ) ) );
-		}
-
-		update_option( 'wp_rocket_settings', $options );
+		return $input;
 	}
 
 	/**
-	 * Restores the delay_js_scripts option to the default value
+	 * Gets a regex pattern of excluded paths for wp-content and wp-includes
 	 *
-	 * @since 3.7
+	 * @since 3.9
 	 *
-	 * @return bool|string
+	 * @return string
 	 */
-	public function restore_defaults() {
-		if ( ! current_user_can( 'rocket_manage_options' ) ) {
-			return false;
+	private function get_excluded_internal_paths() : string {
+		$wp_content = wp_parse_url( content_url(), PHP_URL_PATH );
+		$pattern    = 'wp-includes(.*)';
```

Comment:
What about using `includes_url` function OR at least `WPINC` constant?

Reply:
true it's better since there is a filter there

Reply:
Yes but be careful as this will return the full url for `wp-includes` directory and I don't know if you need also to exclude any external url that has `wp-includes` keyword on it or not?!